gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestPlanMapper.java

```diff
@@ -76,7 +77,15 @@ public MedicationRequest mapToPlanMedicationRequest(RCMRMT030101UKEhrExtract ehr
         if (ehrSupplyAuthoriseIdExtract.isPresent()) {
             var ehrSupplyAuthoriseId = ehrSupplyAuthoriseIdExtract.get();
             var discontinue = extractMatchingDiscontinue(ehrSupplyAuthoriseId, ehrExtract);
-            MedicationRequest medicationRequest = createMedicationRequestSkeleton(ehrSupplyAuthoriseId);
+
+            final Meta meta = confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                MedicationMapperUtils.META_PROFILE,
+                medicationStatement.getConfidentialityCode()
+            );
+
+            final MedicationRequest medicationRequest = (MedicationRequest) new MedicationRequest()
```

Comment:
I think sections "final MedicationRequest medicationRequest = (MedicationRequest) new MedicationRequest()
                .setMeta(meta)
                .setId(ehrSupplyPrescribeId);

            medicationRequest.addIdentifier(buildIdentifier(ehrSupplyPrescribeId, practiseCode));
            medicationRequest.setStatus(COMPLETED);
            medicationRequest.setIntent(ORDER);

            medicationRequest.addDosageInstruction(buildDosage(medicationStatement.getPertinentInformation()));
            medicationRequest.setDispenseRequest(buildDispenseRequestForPrescribe(supplyPrescribe));" from MedicationRequestPlanMapper.java and MedicationRequestOrderMapper.java can be extracted and reused from the common method as they are pretty identical with some minor differences.

Reply:
I am not sure if the order matters but setIntent and setDispenseRequest can be called via fluent API